src/main/java/com/meta/cp4m/message/MessageFactory.java

```diff
@@ -46,7 +46,8 @@ T newMessage(
       Identifier senderId,
       Identifier recipientId,
       Identifier instanceId,
-      Role role);
+      Role role,
+      Message parentMessage);
```

Comment:
Looks like `parentMessage` is always null, we should just eliminate it and you can use `withParent` if needed